code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/UrlUtils.java

```diff
@@ -49,4 +51,39 @@ public static boolean isValidUrl(final String stringUrl) {
     public static String urlEncode(final String unencodedString) {
         return UrlEncoder.urlEncode(unencodedString);
     }
+
+    /**
+     * Extra URL parameters and return as a {@code Map}
+     *
+     * @param queryString the {@code String} of the URL parameter section
+     * @return a {@code Map} of URL parameters
+     */
+    public static Map<String, String> extractQueryParameters(final String queryString) {
```

Comment:
What do you think of using the [android.net.Uri#getQueryParameterNames](https://developer.android.com/reference/android/net/Uri#getQueryParameters(java.lang.String)) and then use the [uri.getQueryParameter(<paramater_name>)](https://developer.android.com/reference/android/net/Uri#getQueryParameter(java.lang.String)) to build the result?  (we may need to have the callers pass the entire url/uri to use it but I think that is just fine) 
That way our code does not have any internal parsing logic on the url/uri.

Reply:
It looks like we don't need to expose this utility method in Core. Other extensions can use the above Android APIs to get the map they need.  @praveek @sbenedicadb thoughts?

Reply:
I think that there is still some value in exposing this utility as it is being using by 2 extensions. My intention was just to ensure the logic does not have additional parsing/splitting..

Reply:
I think having several lines of duplicated logic in extensions should be fine. It should be easier to make changes for the public APIs in Core.